cmd/monaco/download/download_command.go

```diff
@@ -16,140 +16,115 @@ package download
+			if url != "" {
+				manifestFile = ""
+
+				options := directDownloadCmdOptions{
+					environmentURL: url,
+					auth: auth{
+						token:        token,
+						clientID:     oAuthClientID,
+						clientSecret: oAuthClientSecret,
 					},
-					specificAPIs:    specificApis,
-					specificSchemas: specificSettings,
-					onlyAPIs:        onlyAPIs,
-					onlySettings:    onlySettings,
-				},
-			}
-			return command.DownloadConfigs(fs, options)
+					downloadCmdOptions: downloadCmdOptions{
+						sharedDownloadCmdOptions: sharedDownloadCmdOptions{
+							projectName:    project,
+							outputFolder:   outputFolder,
+							forceOverwrite: forceOverwrite,
+						},
+						specificAPIs:    specificApis,
+						specificSchemas: specificSettings,
+						onlyAPIs:        onlyAPIs,
+						onlySettings:    onlySettings,
+					},
+				}
+
+				return command.DownloadConfigs(fs, options)
+
+			} else {
+				if specificEnvironment == "" {
+					return errors.New("missing --environment/-e flag")
+				}
+				options := manifestDownloadOptions{
+					manifestFile:            manifestFile,
+					specificEnvironmentName: specificEnvironment,
+					downloadCmdOptions: downloadCmdOptions{
+						sharedDownloadCmdOptions: sharedDownloadCmdOptions{
+							projectName:    project,
+							outputFolder:   outputFolder,
+							forceOverwrite: forceOverwrite,
+						},
+						specificAPIs:    specificApis,
+						specificSchemas: specificSettings,
+						onlyAPIs:        onlyAPIs,
+						onlySettings:    onlySettings,
+					},
+				}
 
+				return command.DownloadConfigsBasedOnManifest(fs, options)
+			}
 		},
 	}
 
-	setupSharedConfigsFlags(manifestDownloadCmd, &project, &outputFolder, &forceOverwrite, &specificApis, &specificSettings, &onlyAPIs, &onlySettings)
-	setupSharedConfigsFlags(directDownloadCmd, &project, &outputFolder, &forceOverwrite, &specificApis, &specificSettings, &onlyAPIs, &onlySettings)
+	setupSharedConfigsFlags(downloadCmd, &project, &outputFolder, &forceOverwrite, &specificApis, &specificSettings, &onlyAPIs, &onlySettings)
+	downloadCmd.Flags().StringVar(&token, "token", "", "Token secret to connect to DT server")
+	downloadCmd.Flags().StringVar(&oAuthClientID, "oauth-client-id", "", "OAuth client ID is used to connect to DT server via OAuth (mandatory for OAuth access type)")
+	downloadCmd.Flags().StringVar(&oAuthClientSecret, "oauth-client-secret", "", "OAuth client secret is used to connect to DT server via OAuth (mandatory for OAuth access type)")
+	downloadCmd.Flags().StringVarP(&manifestFile, "manifest", "m", "manifest.yaml", "Path to the manifest.yaml file to be read")
+	downloadCmd.Flags().StringVarP(&specificEnvironment, "environment", "e", "", "Specify a concrete environment that shall be downloaded (only usable with --manifest)")
+	downloadCmd.Flags().StringVar(&url, "url", "", "URL to the dynatrace environment from which to download configuration from")
+	downloadCmd.MarkFlagsMutuallyExclusive("url", "manifest")
```

Comment:
either as manual `PreRun` logic or here via `MarkFlagsMutuallyExclusive` we also need to exclude: 

manifest + either oAuth flag

manifest + token should be made impossible by the fact that URL and token are required together, but URL and manifest are exclusive. 
Cobra might give a better error message if token + manifest are already invalid though

Reply:
done i preRunE